module/project/templateTasks/helper.js

```diff
@@ -0,0 +1,638 @@
+                    }
+                }
+
+                let solutionData = {};
+
+                if ( solutionIds.length > 0 ) {
+                    
+                    let solutions = 
+                    await kendraService.solutionDocuments({
+                        externalId : { $in : solutionIds },
+                        isReusable : true
+                    },["externalId"]);
+
+                    if ( 
+                        solutions.success && 
+                        solutions.data &&
+                        Object.keys(solutions.data).length > 0 
+                    ) {
+
+                        solutions.data.forEach(solution => {
+                            if(!solutionData[solution.externalId]) {
+                                solutionData[solution.externalId] = solution._id;
+                            }
+                        });
+                    }
+                }
+
+                return resolve({
+                    tasks : tasks,
+                    templateId : templateId,
+                    solutionData : solutionData
+                });
+
+           } catch (error) {
+               return reject(error);
+           }
+       });
+    }
+
+    /**
+     * Create a task.
+     * @method
+     * @name createOrUpdateTask
+     * @param {Object} data - task data.
+     * @param {String} templateId - template task id
+     * @param {Object} solutionData - solution data
+     * @param {String} [update = false]
+     * @returns {Array} Create or update a task. 
+     */
+    
+    static createOrUpdateTask( 
+        data,
+        templateId,
+        solutionData,
+        update = false
+    ) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let parsedData = UTILS.valueParser(data);
+
+                let allValues = {};
+                allValues.type = parsedData.type.toLowerCase(); 
+
+                if ( allValues.type === CONSTANTS.common.CONTENT ) {
+                    allValues.contentDetails = {};
+
+                    if( parsedData.contentId && parsedData.contentId !== "" ) {
+                        allValues.contentDetails["contentId"] = ObjectId(parsedData.contentId);
+                    } else {
+                        parsedData.STATUS = CONSTANTS.apiResponses.REQUIRED_CONTENT_ID;
+                    }
+
+                    if( parsedData.contentType && parsedData.contentType !== "" ) {
+                        allValues.contentDetails["contentType"] = 
+                        parsedData.contentType;
+                    } else {
+                        parsedData.STATUS = 
+                        CONSTANTS.apiResponses.REQUIRED_CONTENT_TYPE;
+                    }
+
+                } else if ( allValues.type === CONSTANTS.common.IMPROVEMENT_PROJECT ) {   
+                    parsedData.improvementProjectDetails = {};
+
+                    if ( 
```

Comment:
@aman-tunerlabs - I suggest we have a UTILS function which checks this for you instead of writing everywhere everytime, it can work with a single key or an array of keys and object as a parameter.